pandas/core/internals/managers.py

```diff
@@ -1401,7 +1401,7 @@ def insert(self, loc: int, item: Hashable, value: ArrayLike, refs=None) -> None:
         # insert to the axis; this could possibly raise a TypeError
         new_axis = self.items.insert(loc, item)
 
-        if value.ndim == 2:
+        if value.ndim >= 2:
```

Comment:
why is this needed?

Reply:
> why is this needed?

A ndarray where dim >= 2 will bypass the check here, but still raise an Exception when accessing the DataFrame. For an example for this bug, see #53366